src/main/kotlin/no/nav/familie/ef/sak/service/steg/BehandlingSteg.kt

```diff
@@ -0,0 +1,133 @@
+package no.nav.familie.ef.sak.service.steg
+
+import no.nav.familie.ef.sak.repository.domain.Behandling
+import no.nav.familie.ef.sak.repository.domain.BehandlingStatus
+import no.nav.familie.ef.sak.repository.domain.BehandlingType
+
+interface BehandlingSteg<T> {
+
+    fun utførStegOgAngiNeste(behandling: Behandling,
+                             data: T): StegType
+
+    fun stegType(): StegType
+
+
+    fun hentNesteSteg(behandling: Behandling): StegType {
+        return behandling.steg.hentNesteSteg(utførendeStegType = this.stegType(), behandlingType = behandling.type)
+    }
+
+    fun preValiderSteg(behandling: Behandling, stegService: StegService? = null) {}
+    fun postValiderSteg(behandling: Behandling) {}
+
+}
+
+enum class StegType(val rekkefølge: Int,
+                    val tillattFor: List<BehandlerRolle>,
+                    private val gyldigIKombinasjonMedStatus: List<BehandlingStatus>) {
+
+    REGISTRERE_SØKNAD(
+            rekkefølge = 1,
+            tillattFor = listOf(BehandlerRolle.SYSTEM, BehandlerRolle.SAKSBEHANDLER),
+            gyldigIKombinasjonMedStatus = listOf(BehandlingStatus.UTREDES)),
+    VILKÅRSVURDERE_INNGANGSVILKÅR(
+            rekkefølge = 2,
+            tillattFor = listOf(BehandlerRolle.SYSTEM, BehandlerRolle.SAKSBEHANDLER),
+            gyldigIKombinasjonMedStatus = listOf(BehandlingStatus.UTREDES)),
+    VILKÅRSVURDERE_STØNAD(
+            rekkefølge = 3,
+            tillattFor = listOf(BehandlerRolle.SYSTEM, BehandlerRolle.SAKSBEHANDLER),
+            gyldigIKombinasjonMedStatus = listOf(BehandlingStatus.UTREDES)),
```

Comment:
Skal vi ha beregning som et eget steg etter `VILKÅRSVURDERE_STØNAD`? Hos oss må saksbehandler gjøre det manuelt, tror jeg. På barnetrygd blir det gjort automatisk som en del av vilkårsvurderingen, men tror vi er godt tjent med å ha et eget steg for det.

Reply:
Veldig godt poeng, det trenger vi nok helt klart!